docs/src/Cli/GenerateCommand.php

```diff
@@ -142,69 +141,67 @@ private function get_phpdoc_data( $path ) {
 			exit;
 		}
 
-		$parsed_data = $parser->parse_files( $files, $path );
+		$parsed_data = array_map( [ $this, 'filter_internal_data' ], $parser->parse_files( $files, $path ) );
 
-		return array_filter( $parsed_data, [ $this, 'filter_parsed_data' ] );
+		return array_filter(
+			$parsed_data,
+			static function ( $file ) {
+				return ! empty( $file['classes'] )
+				|| ! empty( $file['functions'] )
+				|| ! empty( $file['hooks'] );
+			}
+		);
 	}
 
 	/**
 	 * Filter the parsed data to remove internal and deprecated elements.
 	 *
-	 * @param File $file Individual file to filter.
-	 * @return bool Whether the file should be included or not.
+	 * @param array $file Individual file data to filter.
+	 * @return array File data without internal and deprecated elements.
 	 */
-	private function filter_parsed_data( $file ) {
-		if ( isset( $file['classes'] ) ) {
-			$file['classes'] = array_filter(
-				$file['classes'],
-				[ $this, 'is_not_internal' ]
-			);
+	private function filter_internal_data( $file ) {
+		$file['hooks'] = [];
 
+		if ( isset( $file['classes'] ) ) {
 			foreach ( $file['classes'] as $index => $class ) {
 				if ( isset( $class['methods'] ) ) {
-					$file['classes'][ $index ]['methods'] = array_filter(
-						$class['methods'],
-						[ $this, 'is_not_internal' ]
-					);
-
 					foreach ( $class['methods'] as $method_index => $method ) {
 						if ( isset( $method['hooks'] ) ) {
-							$file['classes'][ $index ]['methods'][ $method_index ]['hooks'] = array_filter(
-								$method['hooks'],
-								[ $this, 'is_not_internal' ]
-							);
+							$file['hooks'] = array_merge( $file['hooks'], $method['hooks'] );
```

Comment:
Assigning the hook to the file breaks the object tree but if the method (or function) is removed due to it being internal, the hook cannot be processed. I'm open to any suggestions on improving this.

Reply:
I think this is good enough for now, to get the documentation into a more correct state.

How to handle these "dependency" issues with hooks can then be revisited when we add the usage information to the pages.